src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/IntermediaryMinecraftProvider.java

```diff
@@ -85,24 +125,26 @@ protected void configureRemapper(RemappedJars remappedJars, TinyRemapper.Builder
 
 	public static final class SingleJarImpl extends IntermediaryMinecraftProvider<SingleJarMinecraftProvider> implements SingleJar {
 		private final SingleJarEnvType env;
+		private final MappingsNamespace sourceNs;
 
-		private SingleJarImpl(Project project, SingleJarMinecraftProvider minecraftProvider, SingleJarEnvType env) {
+		private SingleJarImpl(Project project, SingleJarMinecraftProvider minecraftProvider, SingleJarEnvType env, MappingsNamespace sourceNs) {
 			super(project, minecraftProvider);
 			this.env = env;
+			this.sourceNs = sourceNs;
 		}
 
 		public static SingleJarImpl server(Project project, SingleJarMinecraftProvider minecraftProvider) {
-			return new SingleJarImpl(project, minecraftProvider, SingleJarEnvType.SERVER);
+			return new SingleJarImpl(project, minecraftProvider, SingleJarEnvType.SERVER, minecraftProvider.canMergeJars() ? MappingsNamespace.OFFICIAL : MappingsNamespace.SERVER_OFFICIAL);
 		}
 
 		public static SingleJarImpl client(Project project, SingleJarMinecraftProvider minecraftProvider) {
-			return new SingleJarImpl(project, minecraftProvider, SingleJarEnvType.CLIENT);
+			return new SingleJarImpl(project, minecraftProvider, SingleJarEnvType.CLIENT, minecraftProvider.canMergeJars() ? MappingsNamespace.OFFICIAL : MappingsNamespace.CLIENT_OFFICIAL);
```

Comment:
This shouldnt be needed, as you pass the side specific NS's above.

Reply:
Oh, I see is this for when you want to single jars 🤔 This is going to be a breaking change, as the ns will now change. Not too sure what to do about this.

Reply:
Perhaps Loom can provide a built in intermediate mappings provider that uses the 'modern style' namespaces for old versions?

Reply:
On 2nd thoughts im not too botherd about breaking things like this, this old version support has always been somewhat experimental anyway.